examples/geometry_cube.py

```diff
@@ -25,12 +24,22 @@ def animate():
     # would prefer to do this in a resize event only
     width, height, ratio = canvas.get_size_and_pixel_ratio()
     camera.aspect = width / height
-    camera.update_projection_matrix()
+
+    # cube.rotation.x += 0.005
+    # cube.rotation.y += 0.01
+    rot = vv.linalg.Quaternion().set_from_euler(vv.linalg.Euler(0.005, 0.01))
+    cube.rotation.multiply(rot)
 
     # actually render the scene
     renderer.render(scene, camera)
 
 
+timer = QtCore.QTimer()
+timer.setInterval(1000 * 1 / 30)
+timer.timeout.connect(canvas.update)
```

Comment:
Yes, that's `QWidget.update()` which requests a new draw. And `animate()` gets called on each draw.

Reply:
Idea: expose a cross-gui method for this.

Reply:
> Idea: expose a cross-gui method for this.

I would suggest trying to take some inspiration from `window.requestAnimationFrame` if possible ([docs](https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame)).

It has some nice properties:

- animation is paused when the canvas is not visible
- refresh rate of the GUI environment is matched if possible
- callback is passed the time delta since the last call
- you can request another call yourself; or not!

Reply:
Most of it already works this way. The drawing is scheduled by the GUI (and will probably not draw when hidden). You're right that we can just call `canvas.update()` inside `animate()`, so we can get rid of the timer. Tracking the rest in https://github.com/almarklein/wgpu-py/issues/63